sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CollationTypeCasts.scala

```diff
@@ -47,7 +47,7 @@ object CollationTypeCasts extends TypeCoercionRule {
 
     case otherExpr @ (
       _: In | _: InSubquery | _: CreateArray | _: ArrayJoin | _: Concat | _: Greatest | _: Least |
-      _: Coalesce | _: BinaryExpression | _: ConcatWs) =>
+      _: Coalesce | _: BinaryExpression | _: ConcatWs | _: StringLocate) =>
```

Comment:
We need to special case this expression. Look at @nikolamand-db PR for https://github.com/apache/spark/pull/45963/files example test in CollationSuite. It is important that we do not take into account parameters that should not be strings into this resolution.